servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/auth/BasicAuthHttpServiceFilter.java

```diff
@@ -359,7 +359,7 @@ public Completable closeAsyncGracefully() {
             if (config.userInfoKey != null) {
                 AsyncContext.put(config.userInfoKey, userInfo);
             }
-            return delegate().handle(ctx, request, factory);
+            return delegate().handle(ctx, request, factory).subscribeShareContext();
```

Comment:
`onAuthenticated` method itself doesn't have a context how it is invoked. It's important for the `flatMap`, but may not be important if the caller code changes. Consider moving `subscribeShareContext()` where the `flatMap` is used.